core/src/main/clojure/xtdb/log.clj

```diff
@@ -247,8 +251,53 @@
                        :kafka :xtdb.kafka/log)
                      opts))
 
-(defmethod ig/init-key :xtdb/log [_ ^Log$Factory factory]
-  (.openLog factory))
+(defmethod ig/prep-key :xtdb/log [_ factory]
+  {:block-cat (ig/ref :xtdb/block-catalog)
+   :factory factory})
+
+(defn tx-id->offset [^long tx-id]
```

Comment:
try to re-use the Kotlin one here, save us the risk of the two implementations being out of sync